Source/Diagnostics/BTDiagnostics.cpp

```diff
@@ -159,8 +159,18 @@ BTDiagnostics::ReadParameters ()
     WARPX_ALWAYS_ASSERT_WITH_MESSAGE(m_do_back_transformed_fields, " fields must be turned on for the new back-transformed diagnostics");
     if (m_do_back_transformed_fields == false) m_varnames.clear();
 
-    getWithParser(pp_diag_name, "num_snapshots_lab", m_num_snapshots_lab);
-    m_num_buffers = m_num_snapshots_lab;
+
+    std::vector<std::string> intervals_string_vec = {"0"};
+    bool const num_snapshots_specified = queryWithParser(pp_diag_name, "num_snapshots_lab", m_num_snapshots_lab);
+    bool const intervals_specified = pp_diag_name.queryarr("intervals", intervals_string_vec);
+    if (num_snapshots_specified)
+    {
+        WARPX_ALWAYS_ASSERT_WITH_MESSAGE(!(num_snapshots_specified && intervals_specified),
+            "For back-transformed diagnostics, user should specify either num_snapshots_lab or intervals, not both");
```

Comment:
Isn't it enough to check that `intervals_specified == false` here, since we are already inside an `if (num_snapshots_specified)` block?
```suggestion
        WARPX_ALWAYS_ASSERT_WITH_MESSAGE(!intervals_specified,
            "For back-transformed diagnostics, user should specify either num_snapshots_lab or intervals, not both");
```
I might be wrong, it's always difficult to get this logic right.

Reply:
You are right.  I thought it might be easier to see the logic requires both statements, but your comment points out that being in nested if statements with the two statements so close in the code makes this redunancy unnecessary